src/tactic/mk_iff_of_inductive_prop.lean

```diff
@@ -2,34 +2,41 @@
 Copyright (c) 2018 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl
+-/
+import tactic.core
+/-!
+# mk_iff_of_inductive_prop
 
-Generation function for iff rules for inductives, like for `list.chain`:
+This file defines a tactic that generates `iff` rules for inductive `Prop`s, like for `list.chain`:
 
-    ∀{α : Type*} (R : α → α → Prop) (a : α) (l : list α),
-      chain R a l ↔ l = [] ∨ ∃{b : α} {l' : list α}, R a b ∧ chain R b l ∧ l = b :: l'
+```lean
+∀{α : Type*} (R : α → α → Prop) (a : α) (l : list α),
+  chain R a l ↔ l = [] ∨ ∃{b : α} {l' : list α}, R a b ∧ chain R b l ∧ l = b :: l'
+```
 
 -/
-import tactic.core
-
 namespace tactic
 
 open tactic expr
 
+/-- Given two expressions `e₀` and `e₁`, return the expression `` `(%%e₀ ↔ %%e₁)``. -/
 meta def mk_iff (e₀ : expr) (e₁ : expr) : expr := `(%%e₀ ↔ %%e₁)
 
-meta def select : ℕ → ℕ → tactic unit
+private meta def select : ℕ → ℕ → tactic unit
 | 0 0             := skip
 | 0 (n + 1)       := left >> skip
 | (m + 1) (n + 1) := right >> select m n
 | (n + 1) 0       := failure
 
 /-- `compact_relation bs as_ps`: Produce a relation of the form:
-  R as := ∃ bs, Λ_i a_i = p_i[bs]
-This relation is user visible, so we compact it by removing each `b_j` where a `p_i = b_j`, and
+```lean
+R as := ∃ bs, Λ_i a_i = p_i[bs]
+```
+This relation is user-visible, so we compact it by removing each `b_j` where a `p_i = b_j`, and
 hence `a_i = b_j`. We need to take care when there are `p_i` and `p_j` with `p_i = p_j = b_k`.
 
 TODO: this is copied from Lean's `coinductive_predicates.lean`, export it there.
```

Comment:
This seems to differ from [the `compact_relation` in `meta.coinductive_predicates`](https://github.com/leanprover-community/mathlib/blob/7a624b8892acbef485e3ad041398c77967ca5f64/src/meta/coinductive_predicates.lean#L220).